### DISCUSSION TITLE
DuckDB in Elixir

### DISCUSSION THREAD

**Discussion by matreyes** [duckdb/duckdb#1916] [Q&A]
Hi, Congratulations for this amazing work!.

I'm interested in using DuckDB from the Elixir Language, which is great for stream processing and realtime, but it lacks for Dataframes, Parquet, etc. So DuckDB would be the best match for building data pipelines or RT analytics.

Calling C++ from Elixir / Erlang is doable through NIFs (C functions), but I have never worked with C/C++, so it would take me some time, even if it should be pretty similar to https://github.com/elixir-sqlite/exqlite (SQLite wrapper).

Te other option would be to wait for the ODBC Driver to be mature and to connect through it, But I have no idea if it will affect the performance or if it will have all the API's that are available.

How do you think I should be approaching this? 

Best.

**Comment by hannes** [comment]
Waiting for ODBC is certainly the easiest option if you don't feel comfortable writing this wrapper yourself. Is the ODBC integration there any good?

**Reply by matreyes** [reply]
I will! 
Fortunately, erlang comes with a great ODBC interface, so it should be plug and play.
Best!
